From f77ef601020b63f7578404ad01e7c47ec8e80300 Mon Sep 17 00:00:00 2001
From: me-it-is <188520719+me-it-is@users.noreply.github.com>
Date: Tue, 5 Aug 2025 22:31:53 -0700
Subject: [PATCH] test

---
 src/hub/dataSources/wpilog/wpilogWorker.ts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/hub/dataSources/wpilog/wpilogWorker.ts b/src/hub/dataSources/wpilog/wpilogWorker.ts
index 47ac400..c7f17ab 100644
--- a/src/hub/dataSources/wpilog/wpilogWorker.ts
+++ b/src/hub/dataSources/wpilog/wpilogWorker.ts
@@ -242,6 +242,7 @@ function parseField(key: string, skipMessage = false) {
               }
             } else if (type.startsWith(PHOTON_PREFIX)) {
               let schemaType = type.split(PHOTON_PREFIX)[1];
+              console.log(schemaType);
               log.putPhotonStruct(key, timestamp, record.getRaw(), schemaType);
             } else if (type.startsWith(PROTO_PREFIX)) {
               let schemaType = type.split(PROTO_PREFIX)[1];
+              console.log(schemaType);

-- 
2.50.1

